packages/records/lib/models/records.ts

```diff
@@ -0,0 +1,417 @@
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc.js';
+import { db } from '../db/client.js';
+import type {
+    FormattedRecord,
+    FormattedRecordWithMetadata,
+    GetRecordsResponse,
+    LastAction,
+    ReturnedRecord,
+    UnencryptedRecord,
+    UpsertSummary
+} from '../types.js';
+import { decryptRecordData, decryptRecords, encryptDataRecords } from '../utils/encryption.js';
+import { RECORDS_TABLE } from '../constants.js';
+import { removeDuplicateKey, getUniqueId } from '../helpers/uniqueKey.js';
+import { logger } from '../utils/logger.js';
+import { resultErr, resultOk } from '@nangohq/shared';
+import type { Result } from '@nangohq/shared';
```

Comment:
my goal is to move `Result` to `@nangohq/utils` so records doesn't have to depend on `shared`